lib/stdlib/test/peer_SUITE.erl

```diff
@@ -154,6 +156,30 @@ peer_down_crash(Config) when is_list(Config) ->
         {fail, disconnect_timeout}
     end.
 
+peer_down_crash_tcp(Config) when is_list(Config) ->
+    %% two-way link: "crash" mode
+    {ok, Peer, Node} = peer:start_link(#{name => peer:random_name(?FUNCTION_NAME),
+                                         peer_down => crash,
+                                         connection => 0}),
+    %% verify node started locally
+    ?assertEqual(Node, peer:call(Peer, erlang, node, [])),
+    %% verify there is no distribution connection
+    ?assertEqual([], erlang:nodes(connected)),
+    %% unlink and monitor
+    unlink(Peer),
+    MRef = monitor(process, Peer),
+    %% Make the node go down
+    ok = erpc:cast(Node, erlang, halt, [0]),
+
+    %% since two-way link is requested, it triggers peer to stop
+    receive
+        {'DOWN', MRef, process, Peer, tcp_closed} ->
+            ok
+    after 2000 ->
```

Comment:
You may want to bump this to 5 sec, debug build may sometimes be a bit too slow on that path.